### PR TITLE
Separate system keys from game keys

### PR DESCRIPTION
**Pull request by fmatthew5876**
Fix: #2097

### CODE REVIEW

**Comment by fmatthew5876** [comment]
One thing I haven't done yet. In the log mechanism, we probably should not save system buttons. Otherwise capture and replay will break.

**Comment by fmatthew5876** [comment]
Tested this, F2 works much better now @500fps.

Not entirely sure about whether the FULLSCREEN key should be a game key or a system key. Thoughts?

**Comment by Ghabry** [comment]
Imo it is bad that replays can toggle the fullscreen but for proper playback you need this flag to be set correctly. So it must be this way.

Maybe the fullscreen status reported can be faked but this is more work to get correctly working - You need a real fullscreen flag and a fake fullscreen flag.

File: src/input.h [hunk 1]
```diff
@@ -125,6 +130,41 @@ namespace Input {
...
+	/**
+	 * Gets if a system button is being pressed.
+	 *
+	 * @param button button ID.
+	 * @return whether the button is being pressed.
+	 */
+	bool IsSystemPressed(InputButton button);
+
+	/**
+	 * Gets if a button is starting to being pressed.
```
Comment (carstene1ns) [inline src/input.h:142]: "system"

Comment (fmatthew5876) [reply src/input.h]: Fixed, thanks!

**Pull request merged**